src/server.py

```diff
@@ -69,14 +69,80 @@ def response_deconstructor(response):
     headers_split = [header.split(':', 1) for header in headers]
     headers_dict = {k.lower(): v.strip() for k, v in headers_split}
     return [protocol, status, msg, headers_dict,
-            str(len(body)), str(len(first_pass))]
+            str(len(body))]
+
+
+def request_deconstructor(request):
+    """Deconstruct request into basic components.
+    Return method, protocol and validates host header"""
+    request_msg = request
```

Comment:
I'm not sure why you are binding an additional name to the request. You already have one.